named_decorator.py

```diff
@@ -4,10 +4,88 @@
 from types import FunctionType
 from six import get_function_code
 from six import get_function_defaults
+from six import PY2
 
-from five import code_type_args_for_rename
-from five import get_function_closure
-from five import get_function_globals
+
+if PY2:
+    def get_function_globals(func):
+        """Gets a Python's function's globals in Py2
+        :param func: python function
+        """
+        return func.func_globals
+else:
+    def get_function_globals(func):
+        """Gets a Python's function's globals in Py3
+        :param func: python function
+        """
+        return func.__globals__
+
+
+if PY2:
```

Comment:
I actually like the duplication. Makes things clearer, compared the version I had before with `.insert(1)`. I'll remove the multiple if/else branches though.